src/common/cron.h

```diff
@@ -23,23 +23,128 @@
 #include <ctime>
 #include <iostream>
 #include <string>
+#include <variant>
 #include <vector>
 
+#include "parse_util.h"
 #include "status.h"
+#include "string_util.h"
 
-struct Scheduler {
-  int minute;
-  int hour;
-  int mday;
-  int month;
-  int wday;
+struct CronPattern {
+  using Number = int;
+  using Range = std::pair<int, int>;
 
-  // Whether we use */n interval syntax
-  bool minute_interval = false;
-  bool hour_interval = false;
-  bool mday_interval = false;
-  bool month_interval = false;
-  bool wday_interval = false;
+  struct Interval {
+    int interval;
+  };                                                         // */n
+  struct Any {};                                             // *
+  using Numbers = std::vector<std::variant<Number, Range>>;  // 1,2,3-6,7
+
+  std::variant<Numbers, Interval, Any> val;
+
+  static StatusOr<CronPattern> Parse(std::string_view str, std::tuple<int, int> minmax) {
```

Comment:
I just wondering why std::tuple and not std::pair since we expect only 2 values. + We already use std::pair in this file.

Reply:
The underlying parameter type is tuple in ParseInt, and I don't want to change it now.